lib/bedrock/index.js

```diff
@@ -27,6 +28,54 @@ module.exports = (data) => {
       }
 
       return itemstates
+    },
+
+    calculateBlockStateHashes() {
+
+      const nbt = require('prismarine-nbt')
+      const blockStatesByRuntimeId = {}
+      for (const [stateId, block] of Object.entries(data.blockStates)) {
+
+        const name = `minecraft:${block.name}`
+
+        var states = {}
+        for (const key of Object.keys(block.states).sort()) {
+
+          states[key] = block.states[key]
+        }
+
+        const tag = nbt.comp({
+
+          name: { type: 'string', value: name },
+          states: nbt.comp(states)
+        })
+
+        var s = nbt.writeUncompressed(tag, 'little').toString()
+        const hash = (name === 'minecraft:unknown' ? -2 : fnv1a32(s))
+
+        blockStatesByRuntimeId[hash] = Number(stateId)
+      }
+
+      data.blockStatesByRuntimeId = blockStatesByRuntimeId
```

Comment:
This should be `data.blocksByRuntimeId` and similarly exposed for mcpc equal to `data.blocksByStateId`